### DISCUSSION TITLE
History vanished and XDrip+ apparently reset itself (forgotten about sensor) overnight

### DISCUSSION THREAD

**Discussion by simonpickering** [NightscoutFoundation/xDrip#3144] [Q&A]
Hi,

I woke up this morning having gone to bed with a working system and ~4 day old sensor, to find XDrip+ with no data showing, and waiting for me to pair a sensor (also all the history had vanished.) There was a repeated message in the log "Got exception processing fields in protocol: java.lang.NumberFormatException: For input string...." {insert different gobbledegook strings each time the message repeats}

I've uploaded a log, a reboot fixed the error showing in the log and I then re-paired with the sensor after updating XDrip+ to the latest nightly (nothing to lose at that point), but still no history available.

I just wondered if anyone had seen similar, I'd prefer to avoid it happening again as the system was offline all night looking at how far back those errors went.

Any thoughts appreciated,

Thanks,

Simon

**Comment by Navid200** [comment]
Thanks for posting.

We have an open issue: https://github.com/NightscoutFoundation/xDrip/issues/1280

Would you please tell me if you had the missed reading alert enabled or not?

When this happens, it is like you have uninstalled and reinstalled xDrip as far as the database is concerned.
But, my understanding at this point is that the settings are intact.  Therefore, if you had the missed readings alert set up, you should have got an alert for missed readings.

This is currently our most serious open issue and we need to resolve it.

I am currently running a test to confirm that after the crash, the missed reading alert will trigger.

**Reply by simonpickering** [reply]
Yes, I have missed readings alert enabled. though it doesn't trigger a sound (mainly because I was getting so many missed readings at night that it kept waking me up and I decided the risk of running low was better than needing to silence an alarm).

fwiw I've always had this set, yet this is the first time I've had a "wipe".

I thought perhaps I'd broken the database as I've got it set to never throw data away - despite this I have noticed that one can't scroll back past a certain point.

Is there a way of restoring the data? Not the end of the world, it's the type of motivator I need to put together an analysis app so I can compare food log entries to the data logged in XDrip (via the nightly SQLite backups I have been saving)

**Reply by simonpickering** [reply]
I didn't notice a missed reading alert, though I may have simply swiped it away by default when I woke up this morning - I couldn't see anything in the log as I could only scroll back so far, which is presumably because as you say above XDrip+ thinks it's been uninstalled so that was where the "new" log started.

**Reply by Navid200** [reply]
If you saved your database, you can restore it.
But, if you didn't, unfortunately, there is no way to recover the history.  I'm sorry about this.

I consider this our worst issue because if the user has not saved their database, there is no way to recover the history.  
It's best to have Nightscout set up so that everything is also stored on Nightscout as well.  
<br/>  

If you have the missed reading alert enabled, I would like to see if it has triggered or not.
Please go to the logs and scroll back and see if you can see this:
![Screenshot_20231019-140031](https://github.com/NightscoutFoundation/xDrip/assets/51497406/fe3542bd-2572-4d07-a8da-51b8365d7f33)

**Reply by Navid200** [reply]
If you have daily database backup enabled, would you please tell me what the size of one backup is?

**Reply by simonpickering** [reply]
No the alert didn't trigger - at least I can't find any mention of it on the log screen (a text search function there would probably help, I'm sure there's an issue about that already though.)

The first entry in the "new" log is for xDrip-Daily, telling me it's completed. Strangely the db it created is larger than the one created yesterday.... so presumably it didn't wipe all of it? - looking at the other files in the folder it hasn't grown as much as a typical day would suggest. I'll have a poke at the files later on and see what's left in them

In any case the latest two are138.16MB and 138.01MB in size.

**Reply by Navid200** [reply]
There is a search engine in logs already.

**Reply by simonpickering** [reply]
You're quite right, I didn't spot it. When was that added? /me goes to look
